### PR TITLE
Upgrade Kestrel to target netcoreapp3.0

### PR DESCRIPTION
**Pull request by Tratcher**
#3754

### CODE REVIEW

**Review by natemcmaster — APPROVED** [review]

**Review by halter73 — APPROVED** [review]

**Comment by Tratcher** [comment]
Blocked on Middleware and IIS
https://github.com/aspnet/AspNetCore/pull/4472
https://github.com/aspnet/AspNetCore/pull/4371

**Comment by natemcmaster** [comment]
We're adding required PR checks to ensure the repo compiles before you merge. Can you update this PR by rebasing on master?

```
git fetch origin master:master
git rebase master
```

Alternative: if you plan to 'squash merge' this PR, you can also use a merge to update this PR. This can be easier to manage if you have lots of changes in this PR.
```
git fetch origin master:master
git merge master
```

**Comment by benaadams** [comment]
Boom! 🎉

**Pull request merged**
